Index: src/audioFilters/mixer.cpp
===================================================================
--- src/audioFilters/mixer.cpp	(revision 3342)
+++ src/audioFilters/mixer.cpp	(working copy)
@@ -128,8 +128,8 @@
           }
          else if (in_nfront == 3)
           {
-           matrix[CH_C][CH_L] = clev * LEVEL_3DB;
-           matrix[CH_C][CH_R] = clev * LEVEL_3DB;
+           matrix[CH_C][CH_L] = clev;
+           matrix[CH_C][CH_R] = clev;
           }
         }
        if (in_nrear == 1)
@@ -274,8 +274,8 @@
     {
      if (out_nfront > 1)
       {
-       matrix[CH_LFE][CH_L]  = lfelev * LEVEL_3DB;
-       matrix[CH_LFE][CH_R]  = lfelev * LEVEL_3DB;
+       matrix[CH_LFE][CH_L]  = lfelev;
+       matrix[CH_LFE][CH_R]  = lfelev;
       }
      else
       matrix[CH_LFE][CH_C]  = lfelev;
