Index: src/ffdshow.rc2
===================================================================
--- src/ffdshow.rc2	(revision 3342)
+++ src/ffdshow.rc2	(working copy)
@@ -35,6 +35,7 @@
             VALUE "ProductName", "ffdshow"
             VALUE "ProductVersion", VERSION_STRING "\0"
             VALUE "Build Date", VERSION_BUILD_DATE_TIME "\0"
+            VALUE "SpecialBuild", "CCCP " ACTUAL_BUILD_YEAR "-" ACTUAL_BUILD_MONTH "-" ACTUAL_BUILD_DAY " " ACTUAL_BUILD_TIME "\0"
         END
     END
     BLOCK "VarFileInfo"
Index: src/SubWCRev.conf
===================================================================
--- src/SubWCRev.conf	(revision 3342)
+++ src/SubWCRev.conf	(working copy)
@@ -2,3 +2,7 @@
 #define BUILD_YEAR "$WCDATE=%Y$"
 #define BUILD_MONTH "$WCDATE=%m$"
 #define BUILD_DAY "$WCDATE=%d$"
+#define ACTUAL_BUILD_YEAR "$WCNOW=%Y$"
+#define ACTUAL_BUILD_MONTH "$WCNOW=%m$"
+#define ACTUAL_BUILD_DAY "$WCNOW=%d$"
+#define ACTUAL_BUILD_TIME "$WCNOW=%X$"
Index: update_version.bat
===================================================================
--- update_version.bat	(revision 3342)
+++ update_version.bat	(working copy)
@@ -20,3 +20,7 @@
 @echo #define BUILD_YEAR 2010 >> src\svn_version.h
 @echo #define BUILD_MONTH 1 >> src\svn_version.h
 @echo #define BUILD_DAY 1 >> src\svn_version.h
+@echo #define ACTUAL_BUILD_YEAR 2010 >> src\svn_version.h
+@echo #define ACTUAL_BUILD_MONTH 1 >> src\svn_version.h
+@echo #define ACTUAL_BUILD_DAY 1 >> src\svn_version.h
+@echo #define ACTUAL_BUILD_TIME 00:00:00 >> src\svn_version.h
