Index: src/filters/transform/Mpeg2DecFilter/Mpeg2DecFilter.cpp
===================================================================
--- src/filters/transform/Mpeg2DecFilter/Mpeg2DecFilter.cpp	(revision 2008)
+++ src/filters/transform/Mpeg2DecFilter/Mpeg2DecFilter.cpp	(working copy)
@@ -37,6 +37,7 @@
 #include <moreuuids.h>
 
 #define EPSILON 1e-4
+#define MPEG2ONLY 1
 
 #ifdef REGISTER_FILTER
 
@@ -66,7 +67,7 @@
 const AMOVIESETUP_FILTER sudFilter[] =
 {
 	#ifdef MPEG2ONLY
-	{&__uuidof(CMpeg2DecFilter), L"MPC - MPEG-2 Video Decoder (Gabest)", 0x00600001, countof(sudpPins), sudpPins, CLSID_LegacyAmFilterCategory},
+	{&__uuidof(CMpeg2DecFilter), L"MPC - MPEG-2 Video Decoder (Gabest)", 0x00500001, countof(sudpPins), sudpPins, CLSID_LegacyAmFilterCategory},
 	#else
 	{&__uuidof(CMpeg2DecFilter), L"MPC - MPEG Video Decoder (Gabest)", 0x00600001, countof(sudpPins), sudpPins, CLSID_LegacyAmFilterCategory},
 	#endif
